### PR TITLE
add admin tools as a separate plugin ogdch_admin

### PR DESCRIPTION
**Pull request by sabinem**
this plugin can then be turned on on demand
it is intended to allow solr reindexing and checks on the solr indexing
there is also a tool to check for the content for a package field
this tool is meant to facilitate testing migrations

### CODE REVIEW

**Comment by sabinem** [comment]
The `ogdch_check_field` api route has much room for improvement. I think it may be still useful during the migration, but I don't want to spent a lot of time on it, so it is rather primitive for now.

File: setup.py [hunk 1]
```diff
@@ -53,6 +53,8 @@
     entry_points='''
     [ckan.plugins]
         ogdch_cmd=ckanext.ogdchcommands.plugin:OgdchCommandsPlugin
+        ogdch_admin=ckanext.ogdchcommands.plugin:OgdchAdminPlugin
+        ogdch_migrate=ckanext.ogdchcommands.plugin:OgdchMigrationPlugin
```
Comment (bellisk) [inline setup.py]: This plugin doesn't seem to exist - was it an earlier idea that was removed?

File: README.md [hunk 2]
```diff
@@ -81,20 +103,23 @@ To install ckanext-ogdchcommands:
...
 
-4. Restart CKAN. For example if you've deployed CKAN with Apache on Ubuntu:
+5. Restart CKAN. For example if you've deployed CKAN with Apache on Ubuntu:
 
      sudo service apache2 reload
 
 ## Config Settings
 
-This extension uses the following config options (.ini file)
+This `ogdch_cmd` uses the following config options (.ini file)
```
Comment (bellisk) [inline README.md]:
```suggestion
The plugin `ogdch_cmd` uses the following config options (.ini file)
```

File: ckanext/ogdchcommands/admin_logic.py [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+        rebuild_search_index(package_id=package_id, only_missing=only_missing)
+    except Exception as e:
+        log.error(str(e))
+        log.error(traceback.format_exc())
+        return {
+            'msg': "an error occured",
+            'error': str(e),
+            'traceback': traceback.format_exc()
+        }
+    return "Success: search index was rebuild"
```
Comment (bellisk) [inline ckanext/ogdchcommands/admin_logic.py]:
```suggestion
    return "Success: search index was rebuilt"
```

File: ckanext/ogdchcommands/admin_logic.py [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+    processed_count = 0
+    log.error("search for datasets")
+    while page == 0 or processed_count < result_count:
+        try:
+            log.error("now searching")
+            page = page + 1
+            start = (page - 1) * rows
+            log.error(fq)
+            log.error(rows)
+            log.error(start)
```
Comment (bellisk) [inline ckanext/ogdchcommands/admin_logic.py]: These error logs seem unnecessary - can we get rid of them, please? There are some others in the other functions in this file as well.

**Review by bellisk — APPROVED** [review]

**Pull request merged**